src/js/tabs/pid_tuning.js

```diff
@@ -1415,6 +1407,482 @@ TABS.pid_tuning.initialize = function (callback) {
+                filterDTermSliderUnavailable = false;
+
+                if ($('.pid_filter input[name="gyroLowpassDynMinFrequency"]').val() != Math.round(FILTER_DEFAULT.gyro_lowpass_dyn_min_hz * gyroFilterSliderValue) ||
+                    $('.pid_filter input[name="gyroLowpassDynMaxFrequency"]').val() != Math.round(FILTER_DEFAULT.gyro_lowpass_dyn_max_hz * gyroFilterSliderValue) ||
+                    $('.pid_filter select[name="gyroLowpassDynType"]').val() != FILTER_DEFAULT.gyro_lowpass_type ||
+                    $('.pid_filter input[name="gyroLowpass2Frequency"]').val() != Math.round(FILTER_DEFAULT.gyro_lowpass2_hz * gyroFilterSliderValue) ||
+                    $('.pid_filter select[name="gyroLowpass2Type"]').val() != FILTER_DEFAULT.gyro_lowpass2_type) {
+
+                    $('.tuningFilterSliders .sliderLabels tr:first-child').css('opacity', 0.3);
+                    $('output[name="tuningGyroFilterSlider-number"]').hide();
+                    $('#tuningGyroFilterSlider').val(1);
+                    filterGyroSliderUnavailable = true;
+                } else {
+                    $('.tuningFilterSliders .sliderLabels tr:first-child').css('opacity', '');
+                    $('output[name="tuningGyroFilterSlider-number"]').show();
+                    initGyroFilterSliderPosition();
+                }
+
+                if ($('.pid_filter input[name="dtermLowpassDynMinFrequency"]').val() != Math.round(FILTER_DEFAULT.dterm_lowpass_dyn_min_hz * dtermFilterSliderValue) ||
+                    $('.pid_filter input[name="dtermLowpassDynMaxFrequency"]').val() != Math.round(FILTER_DEFAULT.dterm_lowpass_dyn_max_hz * dtermFilterSliderValue) ||
+                    $('.pid_filter select[name="dtermLowpassDynType"]').val() != FILTER_DEFAULT.dterm_lowpass_type ||
+                    $('.pid_filter input[name="dtermLowpass2Frequency"]').val() != Math.round(FILTER_DEFAULT.dterm_lowpass2_hz * dtermFilterSliderValue) ||
+                    $('.pid_filter select[name="dtermLowpass2Type"]').val() != FILTER_DEFAULT.dterm_lowpass2_type) {
+
+                    $('.tuningFilterSliders .sliderLabels tr:last-child').css('opacity', 0.3);
+                    $('output[name="tuningDTermFilterSlider-number"]').hide();
+                    $('#tuningDTermFilterSlider').val(1);
+                    filterDTermSliderUnavailable = true;
+                } else {
+                    $('.tuningFilterSliders .sliderLabels tr:last-child').css('opacity', '');
+                    $('output[name="tuningDTermFilterSlider-number"]').show();
+                    initDTermFilterSliderPosition();
+                }
+
+                if (filterGyroSliderUnavailable || filterDTermSliderUnavailable) {
+                    $('.sliderFilterDisabled').show();
+                } else {
+                    
+                    $('.sliderFilterDisabled').hide();
+                }
+            }
+
+            function calculateNewPids() {
```

Comment:
`pid_tuning.js` is a behemoth already. I think it would be good to split the actual slider based calculations into a separate file.
Thoughts?

Reply:
Good point. Should I go ahead and do that?

Reply:
Yes please.

Reply:
Done.